framework/di/Instance.php

```diff
@@ -128,7 +128,11 @@ public static function ensure($reference, $type = null, $container = null)
         }
 
         if ($reference instanceof self) {
-            $component = $reference->get($container);
+            try {
```

Comment:
I do not think the impact is significant. Development speed, i.e. clear error message has more weight here.